pkg/controller/statusmanager/pod_status.go

```diff
@@ -175,54 +205,72 @@ func (status *StatusManager) updateNodeStatus(nodeName string, ready bool, reaso
+	now := time.Now()
+	if pod == nil {
+		pod = &corev1.Pod{}
+		err := status.client.Get(context.TODO(), podName, pod)
 		if err != nil {
-			log.Error(err, "Failed to get nodes for condition updating")
+			log.Error(err, fmt.Sprintf("Error getting %s for node condition", operatortypes.NsxNodeAgentContainerName))
 			return
 		}
-		for _, node := range nodes.Items {
-			status.setNodeNetworkUnavailable(node.ObjectMeta.Name, false, "NSXNodeAgent", "Waiting for nsx-node-agent to be created", sharedInfo)
-		}
-		return
 	}
-	pod := corev1.Pod{}
-	for _, pod = range pods.Items {
-		containerStatus := corev1.ContainerStatus{}
-		ready := true
-		messages := ""
-		nodeName := pod.Spec.NodeName
-		for _, containerStatus = range pod.Status.ContainerStatuses {
-			if containerStatus.State.Running == nil {
-				ready = false
-				if containerStatus.State.Waiting != nil {
-					if containerStatus.State.Waiting.Message == "" {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Waiting.Reason)
-					} else {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Waiting.Message)
-					}
-				} else if containerStatus.State.Terminated != nil {
-					if containerStatus.State.Terminated.Message == "" {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Terminated.Reason)
-					} else {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Terminated.Message)
-					}
+	containerStatus := corev1.ContainerStatus{}
+	ready := true
+	// messages is used to store the error message from container status,
+	// tmpMsgs is for the status that pod just restarted in a short time
+	var messages, tmpMsgs string
+	nodeName := pod.Spec.NodeName
+	var startedAt *time.Time
+	var lastStartedAt *time.Time
+	// when pod is pending, its ContainerStatuses is empty
+	if len(pod.Status.ContainerStatuses) == 0 {
+		ready = false
+		messages = fmt.Sprintf("nsx-node-agent not running: %s.", pod.Status.Phase)
+	}
+	for _, containerStatus = range pod.Status.ContainerStatuses {
+		if containerStatus.State.Running == nil {
+			ready = false
+			if containerStatus.State.Waiting != nil {
+				if containerStatus.State.Waiting.Message == "" {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Waiting.Reason)
+				} else {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Waiting.Message)
 				}
-
+			} else if containerStatus.State.Terminated != nil {
+				if containerStatus.State.Terminated.Message == "" {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Terminated.Reason)
```

Comment:
how about we log both Reason and Message? or is it redundant?

Reply:
A sample:
```
Reason:CrashLoopBackOff
Message:back-off 5m0s restarting failed container=nsx-node-agent pod=nsx-node-agent-h452p_nsx-system(c24348c2-ee34-496e-89a1-731429e3c9e0)
```
I think we don't need the reason if we have the detailed message.

Reply:
Cool